### PR TITLE
Extended approval test with drop = True

### PR DESCRIPTION
**Pull request by doctrino**
**Bonus** Found a bug in the TimeSeries configuration.

### CODE REVIEW

File: tests/test_approval_modules.py [hunk 1]
```diff
@@ -109,7 +109,7 @@ def fake_read_yaml_files(
             build_dir=str(local_tmp_path),
             build_env="test",
             interactive=False,
-            drop=False,
+            drop=True,
```
Comment (gregertw) [inline tests/test_approval_modules.py:112]: Do we then always test with drop=True or do we also run the tests with drop=False? A bit hard to say without going outside the code review.

File: tests/conftest.py [hunk 2]
```diff
@@ -64,34 +66,51 @@ def cognite_client_approval() -> CogniteClient:
...
+        client.time_series = create_mock_api(TimeSeriesAPI, TimeSeriesList, written_resources, deleted_resources)
+        client.raw.databases = create_mock_api(RawDatabasesAPI, DatabaseList, written_resources, deleted_resources)
+        client.transformations = create_mock_api(
+            TransformationsAPI, TransformationList, written_resources, deleted_resources
+        )
         client.transformations.schedules = create_mock_api(
-            TransformationSchedulesAPI, TransformationScheduleList, state
+            TransformationSchedulesAPI, TransformationScheduleList, written_resources, deleted_resources
+        )
+        client.data_modeling.containers = create_mock_api(
```
Comment (gregertw) [inline tests/conftest.py:82]: How do we treat deletion of containers and spaces now with the refactoring if there are instances? Will they fail? I believe we originally had that they failed unless --delete-data flag was present, then instances were cleaned out. If we go with my suggestion to only delete data with cdf-tk clean, it makes sense to fail deletion of spaces and containers if there are instances. Our current approach is based on default write and assume nothing exists, thus we need --drop to delete before writing. I assume we will want to change the default behaviour to upsert where there are changes (by comparing yamls). Then --drop will continue to work as today: drop the config before upserting, thus upsert becomes a write.

**Review by gregertw — APPROVED** [review]
Added a couple of comments, but approving.

File: tests/test_approval_modules.py (hunk 1 above)
Comment (doctrino) [reply tests/test_approval_modules.py:112]: Drop is behind a if statement, so I do not think there is much of a value to test with drop=False for this type of test. It was a mistake by me earlier to not set it to true.

**Pull request merged**